deltacat/compute/compactor_v2/utils/compaction_utils.py

```diff
@@ -0,0 +1,713 @@
+        == params.destination_partition_locator.partition_values
+        and rcf_source_partition_locator.stream_id
+        == params.destination_partition_locator.stream_id
+    )
+    if is_inplace_compacted:
+        logger.info(
+            "Overriding round completion file source partition locator as in-place compacted. "
+            + f"Got compacted partition partition_id of {compacted_partition.locator.partition_id} "
+            f"and rcf source partition_id of {rcf_source_partition_locator.partition_id}."
+        )
+        rcf_source_partition_locator = compacted_partition.locator
+
+    round_completion_file_s3_url = rcf.write_round_completion_file(
+        params.compaction_artifact_s3_bucket,
+        rcf_source_partition_locator,
+        compacted_partition.locator,
+        new_round_completion_info,
+        **params.s3_client_kwargs,
+    )
+
+    return ExecutionCompactionResult(
+        compacted_partition,
+        new_round_completion_info,
+        round_completion_file_s3_url,
+        is_inplace_compacted,
+    )
+
+
+def _upload_compaction_result(
+    params: CompactPartitionParams,
+    execute_compaction_result: ExecutionCompactionResult,
+) -> str:
+    compaction_session_type: str = (
+        "INPLACE" if execute_compaction_result.is_inplace_compacted else "NON-INPLACE"
+    )
+    logger.info(
+        f"Partition-{params.source_partition_locator} -> "
+        f"{compaction_session_type} Compaction session data processing completed"
+    )
+    if execute_compaction_result.new_compacted_partition:
+        previous_partition: Optional[Partition] = None
+        if execute_compaction_result.is_inplace_compacted:
+            previous_partition: Optional[
+                Partition
+            ] = params.deltacat_storage.get_partition(
+                params.source_partition_locator.stream_locator,
+                params.source_partition_locator.partition_values,
+                **params.deltacat_storage_kwargs,
+            )
+            # NOTE: Retrieving the previous partition again as the partition_id may have changed by the time commit_partition is called.
+        logger.info(
+            f"Committing compacted partition to: {execute_compaction_result.new_compacted_partition.locator} "
+            f"using previous partition: {previous_partition.locator if previous_partition else None}"
+        )
+        committed_partition: Partition = params.deltacat_storage.commit_partition(
+            execute_compaction_result.new_compacted_partition,
+            previous_partition,
+            **params.deltacat_storage_kwargs,
+        )
+        logger.info(f"Committed compacted partition: {committed_partition}")
+    else:
+        logger.warning("No new partition was committed during compaction.")
+
+    logger.info(f"Completed compaction session for: {params.source_partition_locator}")
+    return execute_compaction_result.round_completion_file_s3_url
```

Comment:
Could we make this function to just `commit_partition` purpose and not returning a `round_completion_files_s3_url`?